collector.go

```diff
@@ -50,3 +50,71 @@ func (c *Collector[T]) Report(f func(report func(T)) error) Task {
 func (c *Collector[T]) Run(f func() T) Task {
 	return NoError(func() { c.report(f()) })
 }
+
+// A Runner manages a group of [Task] that report values.  At least the Go
+// field must be populated.  A Runner must not be copied after first use.
+type Runner[T any] struct {
+	// Go starts a task in a goroutine. This field must be non-nil.
+	// This field is read-only after initialization.
```

Comment:
All fields seem read-only after first use. Maybe move to struct docs? `A Runner must not be copied or its fields changed after first use.` ?

Reply:
Yeah, that is better, thanks.

Reply:
I wound up re-working this to avoid the problem.